packages/fiori/src/NotificationListItemBase.ts

```diff
@@ -0,0 +1,277 @@
+
+/**
+ * Defines if a busy indicator would be displayed over the item.
+ * @type {boolean}
+ * @defaultvalue false
+ * @public
+ * @name sap.ui.webc.fiori.NotificationListItemBase.prototype.busy
+ * @since 1.0.0-rc.8
+ */
+@property({ type: Boolean })
+busy!: boolean;
+
+/**
+ * Defines the delay in milliseconds, after which the busy indicator will show up for this component.
+ *
+ * @type {sap.ui.webc.base.types.Integer}
+ * @defaultValue 1000
+ * @public
+ * @name sap.ui.webc.fiori.NotificationListItemBase.prototype.busyDelay
+ */
+@property({ validator: Integer, defaultValue: 1000 })
+busyDelay!: number;
+
+/**
+ * Defines the actions, displayed in the top-right area.
+ * <br><br>
+ * <b>Note:</b> use the <code>ui5-notification-action</code> component.
+ *
+ * @type {sap.ui.webc.fiori.INotificationAction[]}
+ * @slot
+ * @public
+ * @name sap.ui.webc.fiori.NotificationListItemBase.prototype.actions
+ */
+@slot()
+actions!: Array<HTMLElement & NotificationActionInfo>
+
+	static i18nFioriBundle: I18nBundle;
+
+	static get staticAreaTemplate() {
+		return NotificationOverflowActionsPopoverTemplate;
+	}
+
+	static get staticAreaStyles() {
+		return NotificationOverflowActionsPopoverCss;
+	}
+
+	get hasTitleText() {
+		return !!this.titleText.length;
+	}
+
+	get hasPriority() {
+		return this.priority !== Priority.None;
+	}
+
+	get priorityIcon() {
+		return ICON_PER_PRIORITY[this.priority];
+	}
+
+	get overflowButtonDOM() {
+		return this.shadowRoot!.querySelector(".ui5-nli-overflow-btn")! as HTMLElement; // eslint-disable-line
+	}
+
+	get showOverflow() {
+		return !!this.overflowActions.length;
+	}
+
+	get overflowActions() {
+		if (this.actions.length <= 1) {
+			return [];
+		}
+
+		return this.actionsInfo;
+	}
+
+	get standardActions() {
+		if (this.actions.length > 1) {
+			return [];
+		}
+
+		return this.actionsInfo;
+	}
+
+	get actionsInfo() {
+		return this.actions.map(action => {
+			return {
+				icon: action.icon,
+				text: action.text,
+				press: this._onCustomActionClick.bind(this),
+				refItemid: action._id,
+				disabled: action.disabled ? true : undefined,
+				design: action.design,
+			};
+		});
+	}
+
+	/**
+	 * Event handlers
+	 */
+	_onBtnCloseClick() {
+		this.fireEvent("close", { item: this });
+	}
+
+	_onBtnOverflowClick() {
+		this.openOverflow();
+	}
+
+	_onCustomActionClick(e: MouseEvent) {
+		const refItemId = (e.target as Element).getAttribute("data-ui5-external-action-item-id");
+
+		if (refItemId) {
+			(this.getActionByID(refItemId)! as unknown as UI5Element).fireEvent("click", {
+				targetRef: e.target,
+			}, true);
+
+			this.closeOverflow();
+		}
+	}
+
+	_onkeydown(e: KeyboardEvent) {
+		super._onkeydown(e);
+
+		if (getEventMark(e) === "button") {
+			return;
+		}
+
+		if (isSpace(e)) {
+			e.preventDefault();
+		}
+	}
+
+	getActionByID(id: string) {
+		return this.actions.find(action => action._id === id);
+	}
+
+	async openOverflow() {
+		const overflowPopover = await this.getOverflowPopover();
```

Comment:
Instead of casting to Popover in openOverflow() and closeOverflow(), you can ensure the return type in getOverflowPopover()